upp/stages/resampling.py

```diff
@@ -356,98 +356,148 @@ def run_on_region(self, components: Components, region: Region) -> None:
-                            )
-                        else:
-                            log.warning(
-                                f"[bold yellow]sampling fraction of {sam_frac:.3f}>1 is"
-                                f" needed for component {component}"
-                            )
-
-                    # Ensure the sampling fraction is at least above 0.1
-                    component.sampling_fraction = max(sam_frac, 0.1)
+                # Raise an error/warning if the sampling fraction is above one
+                # for the countup/pdf method
+                if sam_frac > 1:
+                    if self.config.method == "countup":
+                        raise ValueError(
+                            f"[bold red]Sampling fraction of {sam_frac:.3f}>1 is"
+                            f" needed for component {component} This is not supported for"
+                            " countup method."
+                        )
+                    else:
+                        log.warning(
+                            f"[bold yellow]sampling fraction of {sam_frac:.3f}>1 is"
+                            f" needed for component {component}"
+                        )
+
+                # Ensure the sampling fraction is at least above 0.1
+                component.sampling_fraction = max(sam_frac, 0.1)
 
         else:
-            # Set the sampling fraction for each component to the value defined
+            # Set the sampling fraction for the component to the value defined
             # in the config
-            for component in self.components:
-                if component.is_target(self.config.target):
-                    component.sampling_fraction = 1
+            if component.is_target(self.config.target):
+                component.sampling_fraction = 1
+
+            else:
+                component.sampling_fraction = self.config.sampling_fraction
 
-                else:
-                    component.sampling_fraction = self.config.sampling_fraction
+    def run(self, region: str | None = None):
+        """Execute the resampling.
 
-    def run(self):
-        """Execute the resampling."""
+        Parameters
+        ----------
+        region : str | None, optional
+            Define which region is to be resampled, by default None
+            which works through the regions sequentially
+
+        Raises
+        ------
+        ValueError
+            If no region was processed during resampling
+        """
         title = " Running resampling "
         log.info(f"[bold green]{title:-^100}")
         log.info(f"Resampling method: {self.config.method}")
 
-        # Setup the different components and readers/writers
+        # Setup the different components and readers/writers and their sampling fraction
         for component in self.components:
+            # Check if the component is needed for the region
+            if region and region not in component.name:
```

Comment:
Do we want `==` rather than `in` here? or maybe `component.region` or `component.sample`?